charts/datadog/templates/_daemonset-volumes-linux.yaml

```diff
@@ -54,14 +54,45 @@
   name: debugfs
 - name: sysprobe-socket-dir
   emptyDir: {}
-{{- if or .Values.datadog.systemProbe.enableTCPQueueLength .Values.datadog.systemProbe.enableOOMKill }}
+{{- if or .Values.datadog.systemProbe.enableTCPQueueLength .Values.datadog.systemProbe.enableOOMKill .Values.datadog.systemProbe.enableRuntimeCompiler }}
 - hostPath:
     path: /lib/modules
   name: modules
 - hostPath:
     path: /usr/src
   name: src
 {{- end }}
+{{- if .Values.datadog.systemProbe.enableRuntimeCompiler }}
+- hostPath:
+    path: {{ .Values.datadog.systemProbe.runtimeCompilerOutputDir }}
+    type: DirectoryOrCreate
+  name: runtime-compiler-output-dir
+- hostPath:
+    path: {{ .Values.datadog.systemProbe.kernelHeadersDownloadDir }}
+    type: DirectoryOrCreate
+  name: kernel-headers-download-dir
+- hostPath:
+    path: /etc/os-release
+  name: etc-os-release
+- hostPath:
+    path: /etc/redhat-release
+  name: etc-redhat-release
+- hostPath:
+    path: /etc/apt
+  name: apt-config-dir
+- hostPath:
+    path: /etc/yum.repos.d
+  name: yum-repos-dir
+- hostPath:
+    path: /etc/pki
+  name: public-key-dir
+- hostPath:
+    path: /etc/rhsm
+  name: rhel-subscription-dir
+- hostPath:
+    path: /etc/zypp
+  name: opensuse-repos-dir
```

Comment:
the problem with this configuration is that it will created a new folder if it doesn't exist. So user can be bother if the agent deployment create directory on the host.
Also on a harden host, the host file system can be in read only, which prevent to create new folder, and so the agent will be able to start.

Unfortunately kubernetes doesn't provide an option on hostpath to ignore none existing volume...
So I think we should add at least options to:
* disable the mount of all this volume
* (maybe) provide an option to set the path where the linux distribution store the info.

Reply:
> disable the mount of all this volume

We could add a config option for kernel header downloading (separate from runtime compiler config option, but only read if runtime compiler is enabled). We could then disable mounting these volumes if kernel header downloading is disabled. How does that sound?

Reply:
> * provide an option to set the path where the linux distribution store the info.

Can you clarify what you mean by this?

Reply:
@clamoriniere I've added a new flag, `mountPackageManagementDirs`, which allows users to disable the mounting of all the package management directories - does this resolve your concerns?

Reply:
Hi @ISauve,
Yes I think so.